File: exos/20260505-exos-zomboid-gta/05_while.py
# ============================================================
# Hearts of Iron IV — mobilisation nationale
# ============================================================
# La jauge de mobilisation démarre à 10. Chaque semaine, elle
# monte de 8 points. Dès qu'elle atteint ou dépasse 50, la montée
# s'arrête — l'armée est prête. Combien de semaines faut-il ?
#
# Lance : python 05_while.py
# ============================================================

mobilisation = 10
semaines = 0

# Fais monter mobilisation de 8 par semaine jusqu'à ce qu'elle atteigne 50.
# Incrémente semaines à chaque tour de boucle.
#
# départ : mobilisation = 10, semaines = 0
# après 5 tours : mobilisation = 50, semaines = 5
#
# Indice : while + deux += dans le corps de la boucle

while mobilisation < 50:
    mobilisation += 8
    semaines += 1


# --- Vérification (ne pas modifier) ---
assert semaines == 5, (
    f"Obtenu : {semaines} semaines — reprends depuis 10 et ajoute 8 jusqu'à ≥ 50."
)
assert mobilisation >= 50, (
    f"mobilisation vaut {mobilisation} — la boucle doit s'arrêter une fois le seuil atteint."
)
print("✅ Correct !")

File: 20260718/exos/05_while.lua
-- ============================================================
-- Hearts of Iron IV — organisation qui remonte
-- ============================================================
-- Une division a 20 d'orga. Elle récupère 15 par tour tant qu'elle
-- est en dessous de 80. Combien d'orga après la remontée ?
--
-- Lance : lua 05_while.lua
-- ============================================================

orga = 20

-- Tant que orga est strictement inférieur à 80, ajoute 15.
--
-- départ : orga = 20
-- tour 1 : 20 < 80 ? oui  →  orga = 20 + 15 = 35
-- tour 2 : 35 < 80 ? oui  →  orga = 35 + 15 = 50
-- tour 3 : 50 < 80 ? oui  →  orga = 50 + 15 = 65
-- tour 4 : 65 < 80 ? oui  →  orga = 65 + 15 = 80
-- tour 5 : 80 < 80 ? non  →  stop
--
-- Résultat attendu : orga == 80
--
-- Indice : while ... do ... end
--          et orga = orga + 15

-- À toi :
while orga < 80 do
    orga = orga + 15
end

-- --- Vérification (ne pas modifier) ---
assert(orga == 80, "Obtenu : " .. orga .. ", attendu : 80")
print("✅ Correct !")
